apps/api/src/domain/common/events/coscrad-event-consumer.decorator.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { AggregateType } from '@coscrad/api-interfaces';
+import { Injectable } from '@nestjs/common';
+import { ConsoleCoscradCliLogger } from '../../../coscrad-cli/logging';
+import { BaseEvent } from '../../../queries/event-sourcing';
+import buildDummyUuid from '../../models/__tests__/utilities/buildDummyUuid';
+import { dummyDateNow } from '../../models/__tests__/utilities/dummyDateNow';
+import { dummySystemUserId } from '../../models/__tests__/utilities/dummySystemUserId';
+import { CoscradEventConsumer } from './coscrad-event-consumer.decorator';
+import { ICoscradEventHandler } from './coscrad-event-handler.interface';
+import { ICoscradEvent } from './coscrad-event.interface';
+import { SyncInMemoryEventPublisher } from './sync-in-memory-event-publisher';
+
+export interface HasHandles {
+    handles(e: BaseEvent): boolean;
+}
+
+describe('@CoscradEventConsumer', () => {
```

Comment:
We should also consider the following alternative design:
```ts
interface IEventConsumer<TEvent>{
     // this already is on the existing interface
     handle<TEvent extends BaseEvent>(event: TEvent): Promise<void>;

     // we could add this instead of the callback (predicate function) in the decorator factor
     handles(event: BaseEvent): boolean;
}
```

Reply:
We decided to go with the present design,i.e., a callback in the decorator factory. This is because:
1. It makes the overwhelmingly most common case (e.type == "FOO_XED") extremely simple to write
2. It isn't all that hard to debug. You can set a break point in the body of the magic `handles` method and see the predicate run.